### PR TITLE
GreedyMax refactoring - step 2

### PR DESCRIPTION
**Pull request by bryanmiller**
This is based on the dummy optimizer and does not yet handle groups, atoms, or use the visibilities for placing observation. It fixes a few bugs in the dummy optimizer. It does not split, to fill in a night it iterates over observations in order of decreasing score until it finds something to schedule in the remaining time.

### CODE REVIEW

File: app/core/components/ranker/default.py [hunk 1]
```diff
@@ -194,6 +194,12 @@ def _score_obs(self, program: Program, obs: Observation) -> Scores:
                                               np.ones(1, dtype=int) * program.band,
                                               np.ones(1) * 0.8,
                                               program.thesis)
+        print(f'{obs.id:20}  exec: {obs.exec_time().total_seconds() / 3600.:.2f} '\
```
Comment (stroncod) [inline app/core/components/ranker/default.py:197]: This prints should either be in `logging` or removed as is not a good practice to have direct prints in the code.

**Review by stroncod — APPROVED** [review]

**Comment by stroncod** [comment]
There are some minor changes but we could go ahead and merge it.

File: app/core/components/optimizer/greedymax.py [hunk 2]
```diff
@@ -1,10 +1,139 @@
...
+
+
+    def setup(self, program_info: Mapping[ProgramID, ProgramInfo]) -> GreedyMaxOptimizer:
+        """
+        Preparation for the optimizer e.g. create chromosomes, etc.
+        """
+        self.groups = []
+        # self.scores = []
+        for p in program_info.values():
+            self.groups.extend([g for g in p.group_data.values() if g.group.is_observation_group()])
```
Comment (sraaphorst) [inline app/core/components/optimizer/greedymax.py:50]: Do we want to change this now to use the mapping of `UniqueGroupID` to `GroupData` now that it's available?

File: app/core/components/optimizer/greedymax.py [hunk 3]
```diff
@@ -1,10 +1,139 @@
...
+            # Suggestion from Seb
+            # self.groups.extend([g for g in p.group_data.values() if all(x.is_observation_group() for x in g.group.children)])
+            # self.scores.extend([s for s in p.group_data])
+        return self
+
+    def _run(self, plans: Plans):
+
+        print()
```
Comment (sraaphorst) [inline app/core/components/optimizer/greedymax.py:58]: I agree with Sergio that we shouldn't be including `print` statements in PRs that we are going to merge, and should instead be logging.

**Review by sraaphorst — APPROVED** [review]
I'm concerned about the `print` statements being added to the `DefaultRanker`... if they were confined to GreedyMax or were just used for testing prior to merging and then removed, I would feel better, but I understand that we're still tinkering with the scoring a bit so I'll hesitantly approve.

**Pull request merged**